credentialz/credentialz.proto

```diff
@@ -0,0 +1,605 @@
+      returns (stream RotateAccountCredentialsResponse);
+
+  // RotateHostCredentials() allows for changing both the Certificate Authority
+  // public key as well as the keys and the certificate used by the SSH server.
+  // The CA public key key is used to verify certificates presented by
+  // the client to the SSH server during connection establishment.
+  // The server's public and private keys are required during regular operations
+  // of the SSH server.
+  // The SSH server's certificate is used by the client to authenticate
+  // the server during connection establishemnt process. This certificate might
+  // be signed by the same Certificate Authority that has signed clients'
+  // certificates.
+  //
+  // Use Case 1: Update the CA public key.
+  //
+  //   Step 1: Start the stream
+  //     Client <---- RotateHostCredentials() RPC stream begin ------> Target
+  //
+  //   Step 2: Replace the CA public key
+  //     Client -----> CaPublicKeyRequest----> Target
+  //     Client <----- CaPublicKeyResponse <--- Target
+  //
+  //   Step 3 (optional): Test/Validation by the client.
+  //     During this step client attempts to create a new connection to
+  //     the target using the new certificates/keys and validates that
+  //     the certificates/keys "work".
+  //     Once verfied, the client then proceeds to finalize the rotation.
+  //     If the new connection cannot be completed the client will cancel the
+  //     RPC thereby forcing the target to rollback all the certificates/keys.
+  //
+  //   Step 4: Final commit.
+  //     Client -----> FinalizeRequest----> Target
+  //
+  // Use Case 2: Update the server's keys.
+  //
+  //   Step 1: Start the stream
+  //     Client <---- RotateHostCredentials() RPC stream begin ------> Target
+  //
+  //   Step 2: Replace all server keys
+  //     Client -----> ServerKeysRequest----> Target
+  //     Client <----- ServerKeysResponse <--- Target
+  //
+  //   Step 3 (optional): Test/Validation by the client.
+  //     During this step client attempts to create a new connection to
+  //     the target using the new certificates/keys and validates that
+  //     the certificates/keys "work".
+  //     Once verfied, the client then proceeds to finalize the rotation.
+  //     If the new connection cannot be completed the client will cancel the
+  //     RPC thereby forcing the target to rollback all the certificates/keys.
+  //
+  //   Step 4: Final commit.
+  //     Client -----> FinalizeRequest----> Target
+  //
+  rpc RotateHostCredentials(stream RotateHostCredentialsRequest)
+      returns (stream RotateHostCredentialsResponse) {}
+}
+
+//
+// Messages related to RotateAccountCredentials RPC.
+//
+
+message RotateAccountCredentialsRequest {
+  oneof request {
+    AuthorizedKeysRequest credential = 1;
+    AuthorizedUsersRequest user = 2;
+    PasswordRequest password = 3;
+    FinalizeRequest finalize = 4;
+  }
+}
+
+// RotateAccountCredentialsResponse is the response to
```

Comment:
(nit) since you don't have these Go-style comments for each message, you could remove this comment.